src/server/array_func.c

```diff
@@ -137,6 +137,257 @@ static enum job_atr attrs_to_copy[] = {
+		else
+			return 1;
+		for (j = rarr->count - 1; j > i; j--) { /* move the pointers right */
+			rarr->ranges[j] = rarr->ranges[j - 1];
+		}
+		/* now we have space for the index in between */
+		rarr->ranges[i].start = val;
+		rarr->ranges[i].end = val;
+		return 0;
+	}
+
+	/* The value that needs to be added is found between left and right sub-ranges  */
+ 
+	for (i = 0; i < rarr->count; i++) {
+
+		if ((val >= rarr->ranges[i].start) && (val <= rarr->ranges[i].end)) {
+			return 1;
+		}
+
+		if ( (i-1) != -1 && (val > rarr->ranges[i - 1].end) && (val < rarr->ranges[i].start)) {
+			if ((i-1) != -1  && (val == (rarr->ranges[i - 1].end + rarr->step)) && (val == (rarr->ranges[i].start - rarr->step))){
+				/* Adding this value would coalesce these two sub-ranges  */
+				rarr->ranges[i-1].end = rarr->ranges[i].end;
+				for (j = i; j < rarr->count - 1; j++)  /* move the pointers left  (assinging jth = j+1th)  */
+					rarr->ranges[j] = rarr->ranges[j + 1];
+				rarr->count--;
+				rarr->ranges = realloc(rarr->ranges, sizeof(range_t) * rarr->count);
+				return 0;
+			}
+			else if ( (i-1) != -1 &&  (val == (rarr->ranges[i - 1].end + rarr->step)) ) {
+			/* value falls in the left sub-range end  */
+				rarr->ranges[i - 1].end += rarr->step;
+				return 0;
+			} else if (val == (rarr->ranges[i].start - rarr->step)) {
+			/* value falls in the right sub-range start  */
+				rarr->ranges[i].start -= rarr->step;
+				return 0;
+			} else {
+			/* value falls in this range, split range */
+				if ((val - rarr->ranges[i - 1].end) % rarr->step != 0)
+					return 1;
+				rarr->count++;
+				new = realloc(rarr->ranges, sizeof(range_t) * rarr->count);
+				if (new)
+					rarr->ranges = new;
+				else 
+					return 1;
+				for (j = rarr->count - 1; j > i; j--) { /* move the pointers right (assiging j-1th = jth)  */
+					rarr->ranges[j] = rarr->ranges[j - 1];
+				}
+				/* now we have space for the index in between */
+				rarr->ranges[i].start = val;
+				rarr->ranges[i].end = val;
+				return 0;
+			}
+		}
+	}
+
+
+	/* Comming out of loop and check the extreme right corner case */
+	--i;  /* resetting the last inc to match the last sub-range */
+	if ( val > rarr->ranges[i].end) {
+		if (val == rarr->ranges[i].end + rarr->step) {
+			rarr->ranges[i].end += rarr->step;
+			return 0;
+		}	
+		else {
+			rarr->count++;
+			new = realloc(rarr->ranges, sizeof(range_t) * rarr->count);
+			if (new)
+				rarr->ranges = new;
+			else
+				return 1;
+			/* now we have space for the index in between */
+			rarr->ranges[i].start = val;
+			rarr->ranges[i].end = val;
+			return 0;
+		}
+	}
+
+	return 1;
+}
+
+/**
+ * @brief
+ * 			del_value - Deleting a value in the range structures
+ * 						handles adding value in mid of the ranges.
+ *
+ * @param[in]	rarr - pointer to the range_arr_t 
+ * @param[in]	val - Value to delete
+ *
+ * @return      Int
+ * @retval	 0  - Intialization success
+ * @retval	 1  - failure 
+ */
+
+int
+del_value(range_arr_t *rarr, int val)
```

Comment:
we can merge this with range_del_value() and use it in both server and sched. for sched, sched never calls range_del_value to delete any value in middle of range, as sched runs lowest index always (@bhroam please correct me if I am wrong)